.github/workflows/on_commit.yml

```diff
@@ -30,47 +50,30 @@ jobs:
           west config manifest.path sidewalk
           west update -n -o=--depth=1
 
-      - name: twister build samples 
-        run: /workdir/zephyr/scripts/twister --testsuite-root /workdir/sidewalk/samples/${{ matrix.samples_to_test }} --no-clean --show-footprint --footprint-from-buildlog -vvv --build-only
+      - name: Twister build samples 
+        run: /workdir/zephyr/scripts/twister --testsuite-root /workdir/sidewalk/samples/ --no-clean --show-footprint --footprint-from-buildlog -vvv --build-only --subset ${{matrix.subset}}/${{ strategy.job-total }}
       
       - name: Upload build artifacts
         uses: actions/upload-artifact@v3
         with:
-          name: sample-${{ matrix.samples_to_test }}
+          name: sample-${{matrix.subset}}_of_${{ strategy.job-total }}
           path: |
+            twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/merged.hex
             twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/zephyr.elf
             twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/zephyr.hex
             twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/dfu_application.zip
```

Comment:
Please add also or instead of dfu_application.zip, app_update.bin

Reply:
done